hazelcast/src/test/java/com/hazelcast/cache/CacheCreationTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache;
+
+import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.XmlConfigBuilder;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.*;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.cache.Cache;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class CacheCreationTest extends HazelcastTestSupport {
+
+    static Config hzConfig;
+    private static final int THREAD_COUNT = 4;
+
+    @BeforeClass
+    public static void init() throws Exception{
+        final URL configUrl1 = CacheCreationTest.class.getClassLoader().getResource("test-hazelcast-real-jcache.xml");
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(configUrl1.getFile());
+        hzConfig = configBuilder.build();
+    }
+
+    @Before
+    @After
+    public void killAllHazelcastInstances() throws Exception {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void creatingASingleCache() throws URISyntaxException {
+        HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+        Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache" + 1);
+        cache.get(1);
+    }
+
+    @Test
+    public void creatingASingleCacheFromMultiProviders() throws URISyntaxException, InterruptedException {
+
```

Comment:
@pveentjer 
1 - Do you mean at line 69, right ?  Removed.
2 - I'm not sure that I understand - if I remove the threads completely - what would it kept here?

We have just threes tests here:
1:
```@Test
    public void creatingASingleCache() throws URISyntaxException {......} ```

2:
``` @Test
    public void creatingASingleCacheFromMultiProviders() throws URISyntaxException, InterruptedException {}```

3:
``` @Test
    public void creatingMultiCacheFromMultiProviders() throws URISyntaxException, InterruptedException {}```

all those tests for a bit different goals: for single cache, for single + thread count from multi providers (```.getCache("xmlCache");```) and third one for multi cache + + thread count from multi providers (```.getCache("xmlCache"+finalI);```). Honestly the initial idea was from @Danny-Hazelcast and maybe he wants to add something but I said how I understand it - correct me if I'm wrong.

Thanks.